### PR TITLE
Support preview Windows SDK Ref packs, and simplify how they're defined

### PR DESCRIPTION
**Pull request by dsplaisted**
- Instead of duplicating Windows SDK Reference `KnownFrameworkReference` items for each target framework, they can be defined in BundledVersions.props like the following:

```xml
<WindowsSdkSupportedTargetPlatformVersion Include="10.0.19041.0" WindowsSdkPackageVersion="10.0.19041.16" MinimumNETVersion="5.0" />
<WindowsSdkSupportedTargetPlatformVersion Include="10.0.18362.0" WindowsSdkPackageVersion="10.0.18362.16" MinimumNETVersion="5.0" />
<WindowsSdkSupportedTargetPlatformVersion Include="10.0.17763.0" WindowsSdkPackageVersion="10.0.17763.16" MinimumNETVersion="5.0" />
<WindowsSdkSupportedTargetPlatformVersion Include="8.0" />
<WindowsSdkSupportedTargetPlatformVersion Include="7.0" />
```

- Support `UseWindowsSDKPreview` property to opt in to preview Windows SDKs
- Support `WindowsSdkPackageVersion` to explicitly set the Windows SDK Ref package version

### CODE REVIEW

**Comment by dsplaisted** [comment]
> Is there a reason the package versions here are *.16? The latest .NET SDK should be on *.17

@angelazhangmsft Those definitions are just temporary.  The real numbers will come from BundledVersions.props generated in the dotnet/installer repo, but we can't atomically commit to two different repos :-)

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets [hunk 1]
```diff
@@ -17,10 +17,35 @@ Copyright (c) .NET Foundation. All rights reserved.
...
   <UsingTask TaskName="ProcessFrameworkReferences" AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
   <UsingTask TaskName="ResolveAppHosts" AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
 
+  <Target Name="AddWindowsSdkKnownFrameworkReferences"
+          Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' And '$(TargetPlatformIdentifier)' == 'Windows'">
+
+    <!-- Remove Windows SDK KnownFrameworkReference items from BundledVersions.props (they will eventually be removed, but that is in a different repo so
+         we can't do the change atomically -->
+    <ItemGroup>
+      <KnownFrameworkReference Remove="Microsoft.Windows.SDK.NET.Ref" />
```
Comment (joeloff) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets:31]: I take it there are other items that need to remain or would ```Remove="@(KnownFrameworkReference)"``` be better to clear the whole item group?

File: src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAWindowsDesktopProject.cs [hunk 2]
```diff
@@ -342,6 +345,60 @@ public void SupportedOSPlatformVersionCanBeSetInDirectoryBuildTargets()
         }
 
 
+        [WindowsOnlyTheory]
+        //  Basic Windows TargetFramework
+        [InlineData("net5.0-windows10.0.19041.0", false, null, "10.0.19041.*")]
+        //  Basic UseWindowsSdkPreview usage
+        [InlineData("net5.0-windows10.0.99999.0", true, null, "10.0.99999-preview")]
```
Comment (joeloff) [inline src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAWindowsDesktopProject.cs:352]: Do these packages ever use full semantic versioning with build numbers and metadata? It sounded from the comments in the build task that the patterns are fixed, e.g. you add "preview" and that will never change.

File: src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAWindowsDesktopProject.cs [hunk 3]
```diff
@@ -11,9 +11,12 @@
 using System.Xml.Linq;
 using System.IO;
 using System.Linq;
+using System;
 
 namespace Microsoft.NET.Build.Tests
 {
+    //  Add Windows SDK tests here?
```
Comment (wli3) [inline src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAWindowsDesktopProject.cs:18]: ?

**Review by wli3 — APPROVED** [review]

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets (hunk 1 above)
Comment (dsplaisted) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets:31]: Yes, there are other items for all of the shared frameworks we have.

File: src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAWindowsDesktopProject.cs (hunk 2 above)
Comment (dsplaisted) [reply src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAWindowsDesktopProject.cs:352]: Yes, the release versions of the packages use 4 digit build numbers, but NuGet doesn't allow you to use prerelease information in a package version if it has 4 numbers.  So the preview packages will only have 3 numbers, and they plan to always just have them marked as `-preview`.  If they need something else, then the `WindowsSdkPackageVersion` property can be used to explicitly specify the full package version.

File: src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAWindowsDesktopProject.cs (hunk 3 above)
Comment (dsplaisted) [reply src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAWindowsDesktopProject.cs]: oops!

**Review by joeloff — APPROVED** [review]

**Pull request merged**